src/MQTTAsyncUtils.c

```diff
@@ -2320,6 +2320,11 @@ thread_return_type WINAPI MQTTAsync_receiveThread(void* n)
 					m->c->connected = 0; /* don't send disconnect packet back */
 					nextOrClose(m, discrc, "Received disconnect");
 				}
+				else
+				{
+					free(pack);
```

Comment:
I believe, that is not sufficient.
Because `pack` points to many different structures.
E.g. if allocated inside `MQTTPacket_publish(int MQTTVersion, unsigned char aHeader, char* data, size_t datalen)`

pack points to this structure:

```c
/**
 * Data for a publish packet.
 */
typedef struct
{
	Header header;	/**< MQTT header byte */
	char* topic;	/**< topic string */
	int topiclen;
	int msgId;		/**< MQTT message id */
	char* payload;	/**< binary payload, length delimited */
	int payloadlen;	/**< payload length */
	int MQTTVersion;  /**< the version of MQTT */
	MQTTProperties properties; /**< MQTT 5.0 properties.  Not used for MQTT < 5.0 */
	uint8_t mask[4]; /**< the websockets mask the payload is masked with, if any */
} Publish;
```

Which includes pointers to other allocated memory, which is not release in this case.
See issue #1518